// react-emails/lib/Email.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _AlertLayout = require('./layouts/AlertLayout');

var _AlertLayout2 = _interopRequireDefault(_AlertLayout);

var _DownloadLayout = require('./layouts/DownloadLayout');

var _DownloadLayout2 = _interopRequireDefault(_DownloadLayout);

var _ResetPasswordLayout = require('./layouts/ResetPasswordLayout');

var _ResetPasswordLayout2 = _interopRequireDefault(_ResetPasswordLayout);

var _UserCreatedLayout = require('./layouts/UserCreatedLayout');

var _UserCreatedLayout2 = _interopRequireDefault(_UserCreatedLayout);

var _LangDigestLayout = require('./layouts/LangDigestLayout');

var _LangDigestLayout2 = _interopRequireDefault(_LangDigestLayout);

var _LangTextLayout = require('./layouts/LangTextLayout');

var _LangTextLayout2 = _interopRequireDefault(_LangTextLayout);

var _Wrapper = require('./elements/Wrapper');

var _Wrapper2 = _interopRequireDefault(_Wrapper);

var _Wrapper3 = require('./lang/Wrapper');

var _Wrapper4 = _interopRequireDefault(_Wrapper3);

var _types = require('./types');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var keyValueShape = _propTypes2.default.shape({
  key: _propTypes2.default.string,
  value: _propTypes2.default.number
});

var messageShape = _propTypes2.default.shape({
  link: _propTypes2.default.string,
  text: _propTypes2.default.string,
  name: _propTypes2.default.string,
  username: _propTypes2.default.string,
  followers: _propTypes2.default.number
});

var Email = function (_React$Component) {
  _inherits(Email, _React$Component);

  function Email() {
    _classCallCheck(this, Email);

    return _possibleConstructorReturn(this, (Email.__proto__ || Object.getPrototypeOf(Email)).apply(this, arguments));
  }

  _createClass(Email, [{
    key: 'getChildContext',
    value: function getChildContext() {
      var data = this.props.data;


      return {
        brand: {
          name: data.brand_name,
          slug: data.brand_slug
        },
        feed: {
          name: data.feed_name
        },
        product: {
          type: (data.product_type || '').toLowerCase(),
          name: data.product_name,
          slug: data.product_slug
        },
        alert: {
          name: data.alert_name,
          type: data.alert_type,
          threshold: data.threshold,
          interval: data.interval,
          category_name: data.category_name,
          source: data.source,
          link: data.alert_link,
          timezone: data.timezone
        },
        mentions_filters: data.mentions_filters,
        messages: data.messages,
        categories: data.categories,
        concepts: data.concepts,

        download: {
          numMessages: data.numMessages,
          source: data.source,
          startDate: data.startDate,
          endDate: data.endDate,
          productType: data.productType,
          productName: data.productName,
          downloadLink: data.downloadLink
        },

        token: data.token,

        user: {
          email: data.user && data.user.email
        },

        histogram: data.histogram,

        intents: data.intents,

        title: data.title,
        bodyPlainText: data.bodyPlainText,
        bodyHTML: data.bodyHTML
      };
    }
  }, {
    key: 'render',
    value: function render() {
      var type = this.props.type;


      if ([_types.LANG_DIGEST, _types.LANG_TEXT].includes(type)) {
        return _react2.default.createElement(
          _Wrapper4.default,
          { type: type },
          this.layout
        );
      }

      return _react2.default.createElement(
        _Wrapper2.default,
        { type: type },
        this.layout
      );
    }
  }, {
    key: 'layout',
    get: function get() {
      var type = this.props.type;


      switch (type) {
        case _types.ALERT:
          return _react2.default.createElement(_AlertLayout2.default, null);

        case _types.DOWNLOAD_MENTIONS:
          return _react2.default.createElement(_DownloadLayout2.default, null);

        case _types.PASSWORD_RESET:
          return _react2.default.createElement(_ResetPasswordLayout2.default, null);

        case _types.USER_CREATED:
          return _react2.default.createElement(_UserCreatedLayout2.default, null);

        case _types.LANG_DIGEST:
          return _react2.default.createElement(_LangDigestLayout2.default, null);

        case _types.LANG_TEXT:
          return _react2.default.createElement(_LangTextLayout2.default, null);

        default:
          return _react2.default.createElement(
            'p',
            null,
            'Unknown ',
            type,
            ' layout'
          );
      }
    }
  }]);

  return Email;
}(_react2.default.Component);

Email.propTypes = {
  type: _propTypes2.default.string.isRequired,
  data: _propTypes2.default.object.isRequired
};
Email.childContextTypes = {
  brand: _propTypes2.default.shape({
    name: _propTypes2.default.string,
    slug: _propTypes2.default.string
  }),

  feed: _propTypes2.default.shape({
    name: _propTypes2.default.string
  }),

  product: _propTypes2.default.shape({
    type: _propTypes2.default.string,
    name: _propTypes2.default.string,
    slug: _propTypes2.default.string
  }),

  // alert specific stuff
  alert: _propTypes2.default.shape({
    name: _propTypes2.default.string,
    type: _propTypes2.default.string,
    threshold: _propTypes2.default.number,
    interval: _propTypes2.default.string,
    link: _propTypes2.default.string,
    source: _propTypes2.default.string
  }),

  mentions_filters: _propTypes2.default.object,

  messages: _propTypes2.default.arrayOf(messageShape),
  timezone: _propTypes2.default.string,

  // Volume alerts
  categories: _propTypes2.default.arrayOf(keyValueShape),
  concepts: _propTypes2.default.arrayOf(keyValueShape),

  download: _propTypes2.default.shape({
    numMessages: _propTypes2.default.number,
    source: _propTypes2.default.string,
    startDate: _propTypes2.default.string,
    endDate: _propTypes2.default.string,
    productType: _propTypes2.default.string,
    productName: _propTypes2.default.string,
    downloadLink: _propTypes2.default.string
  }),

  // Password reset and user creation
  user: _propTypes2.default.shape({
    email: _propTypes2.default.string
  }),

  token: _propTypes2.default.string,

  histogram: _propTypes2.default.arrayOf(_propTypes2.default.object),

  // Lang.ai digest
  intents: _propTypes2.default.object,

  // Plaintext email
  title: _propTypes2.default.string,
  bodyPlainText: _propTypes2.default.string,
  bodyHTML: _propTypes2.default.string
};
exports.default = Email;
