// react-emails/lib/mocks/alert.js
'use strict';

var dummyMessage = {
  link: 'http://sentisis.io/message-link',
  text: 'Hommage à Chuck Berry, guitariste de génie, interprète d’enfer et parolier inspirant, qui s’est éteint le 18 mars. Il avait 90 ans. Avec lui disparaît une certaine idée du rock’n’roll, spectaculaire et virtuose, qui a permis l’éclosion de nombre de talents dans les années 1960. Retour sur le parcours d’un rockeur pionnier. http://bit.ly/2mMNJ29',
  name: '• LOS GATOS •',
  username: 'losgatos',
  followers: 12312,
  source: 'tw',
  date: "2017-06-13T11:06:35.000Z",

  // Twitter only fields
  reach: 1234,
  retweets: 10,

  // YT only fields
  // views: 145829,

  // Facebook + Instagram (likes is also on Twitter and Youtube)
  likes: 4594
  // comments: 1201,
};

var dummyMessage2 = {
  link: 'http://sentisis.io/message-link',
  text: 'Message 3',
  name: '• LOS GATOS •',
  username: 'losgatos',
  followers: 456,
  source: 'tw',
  date: "2017-06-13T11:06:35.000Z",

  // Twitter only fields
  reach: 66567,
  retweets: 670,

  // YT only fields
  // views: 145829,

  // Facebook + Instagram (likes is also on Twitter and Youtube)
  likes: 4594
  // comments: 1201,
};

var dummyCategoriesOrConcepts = [{ key: 'Quejas por producto', value: 971 }, { key: 'Atención al cliente', value: 627 }, { key: 'Transferecia', value: 254 }, { key: 'Empresa', value: 627 }, { key: 'Quejas por sección', value: 254 }];

module.exports = {
  alert_name: 'Alert name',
  alert_type: 'INFLUENCE',
  alert_user_name: 'user Name',
  alert_link: 'https://twitterc.com',
  brand_name: 'BrandName',
  feed_name: 'FeedName',
  product_name: 'ProdName',
  product_type: 'DRIVE',
  sentiment: 'negative',
  category_name: 'CategoryName',
  product_slug: 'some-slug',
  brand_slug: 'brand-slug',
  handle: 'pedro',
  threshold: 10,
  interval: 'D',
  tracking_pixel_url: 'http://api.mixpanel.com/track/?data=eyJldmVudCI6IlVzZXIgb3BlbnMgYWxlcnQgZW1haWwiLCJwcm9wZXJ0aWVzIjp7ImRpc3RpbmN0X2lkIjoidGhlLnVzZXJAZG9tYWluLmNvbSIsIiRlbWFpbCI6InRoZS51c2VyQGRvbWFpbi5jb20iLCJhbGVydF90eXBlIjoiSU5GTFVFTkNFX05FR0FUSVZFIiwic2VudF90aW1lIjoiMTk3MC0wMS0wMVQwMDowMDowMC4wMDBaIiwidG9rZW4iOiIyYzg2NjZiNmU2YzliZTQxODE1ZDYxMzIyNWJkMTAzZCIsImVtYWlsX2lkIjoiZW1haWxJZCIsIlVzZXIgVHlwZSI6IkN1c3RvbWVyIn19&ip=1&img=1',
  messages: [dummyMessage, dummyMessage2],
  timezone: 'Europe/Madrid',
  categories: dummyCategoriesOrConcepts,
  concepts: dummyCategoriesOrConcepts,
  mentions_filters: {
    author: { or: ['pedro'] },
    from: 321421,
    to: 345345123,
    category: { or: ['56ec00ad214872e717e4b14d', '56ec00ad214872e717e4b145'] }
  },
  "histogram": [{
    "humanKey": "2017-07-04T00:00:00.000+02:00",
    "timestamp": 1499119200000,
    "count": 3,
    "sentiment": {
      "OBJECTIVE": 3
    }
  }]
};
