// react-emails/server/createHTML.js
'use strict';

const fs = require('fs');
const Path = require('path');

const tags = {
  content: '%CONTENT%',
  styles: '%STYLES%',
  subject: '%SUBJECT%',
  tracking: '%TRACKING%',
};


function getFile(p) {
  return new Promise((resolve, reject) => {
    const path = Path.join(__dirname, p);

    return fs.readFile(path, { encoding: 'utf8' }, (err, file) => {
      if (err) return reject(err);
      return resolve(file);
    })
  });
}

function createHTML(content, options) {
  const subject = options.subject || 'Séntisis';

  return Promise.all([
    getFile('../src/css/style.css'),
    getFile('./email.html'),
  ])
    .then((res) => {
      const styles = res[0];
      let template = res[1];

      template = template.replace(tags.subject, subject);
      template = template.replace(tags.styles, styles);
      template = template.replace(tags.content, content);

      let trackingImg = '';
      if (options.trackingSrc) {
        trackingImg = `<img src="${options.trackingSrc}" />`;
      }

      template = template.replace(tags.tracking, trackingImg);

      return template;
    });
}

module.exports = createHTML;
