// react-emails/lib/elements/AlertHeader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _fp = require('lodash/fp');

var _fp2 = _interopRequireDefault(_fp);

var _Img = require('./Img');

var _Img2 = _interopRequireDefault(_Img);

var _Grid = require('./Grid');

var _Grid2 = _interopRequireDefault(_Grid);

var _locale = require('../utils/locale');

var _locale2 = _interopRequireDefault(_locale);

var _format = require('../utils/format');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var alertIconSrc = 'https://s3-eu-west-1.amazonaws.com/sentisis-images/emails/alert-icon.png';

var styles = {
  wrapper: {
    marginBottom: '30px',
    backgroundColor: '#FFF',
    display: 'block',
    borderRadius: '5px'
  },
  innnerTable: {
    padding: '25px'
  },
  Title: {
    fontWeight: 'normal',
    lineHeight: '140%',
    margin: '10px 0 0 0',
    fontSize: '16px'
  },
  icon: {
    verticalAlign: 'top',
    paddingRight: '10px'
  },
  alertTitle: {
    display: 'block',
    fontWeight: 'bold',
    paddingTop: '5px',
    fontSize: '13px'
  }
};

var AlertHeader = function (_React$PureComponent) {
  _inherits(AlertHeader, _React$PureComponent);

  function AlertHeader() {
    _classCallCheck(this, AlertHeader);

    return _possibleConstructorReturn(this, (AlertHeader.__proto__ || Object.getPrototypeOf(AlertHeader)).apply(this, arguments));
  }

  _createClass(AlertHeader, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        _Grid2.default,
        { style: styles.wrapper },
        _react2.default.createElement(
          _Grid2.default.Cell,
          { style: styles.innnerTable },
          _react2.default.createElement(
            _Grid2.default,
            null,
            _react2.default.createElement(
              _Grid2.default.Row,
              null,
              _react2.default.createElement(
                _Grid2.default.Cell,
                { style: styles.icon },
                _react2.default.createElement(_Img2.default, { width: '24px', height: '24px', alt: 'Alert icon', src: alertIconSrc })
              ),
              _react2.default.createElement(
                _Grid2.default.Cell,
                null,
                _react2.default.createElement(
                  'span',
                  { style: styles.alertTitle },
                  'Alerta ',
                  this.alertType
                ),
                _react2.default.createElement('p', { style: styles.Title, dangerouslySetInnerHTML: { __html: this.title } })
              )
            )
          )
        )
      );
    }
  }, {
    key: 'alertType',
    get: function get() {
      var alert = this.context.alert;

      if (alert.type.includes('VOLUME')) return 'por Volumen';
      if (alert.type.includes('AUTHOR')) return 'por Autor';
      if (alert.type.includes('INFLUENCE')) return 'por Influencia';
      return;
    }
  }, {
    key: 'title',
    get: function get() {
      var _context = this.context,
          alert = _context.alert,
          feed = _context.feed,
          _context$messages = _context.messages,
          messages = _context$messages === undefined ? [] : _context$messages;

      var source = '';

      if (messages.length && alert.source !== 'all') {
        source = 'en ' + (0, _locale2.default)('source', messages[0].source);
      }

      var atUsername = ['tw', 'ig'].includes(messages.length && messages[0].source) ? '@' : '';
      var category = alert.category_name ? 'sobre <b>' + alert.category_name + '</b>' : '';
      var interval = _fp2.default.getOr('', alert.interval, {
        M: 'el último minuto',
        H: 'la última hora',
        D: 'el último día'
      });
      var msg = messages.length ? messages[0] : [];
      var threshold = (0, _format.formatNumber)(alert.threshold, { shorten: false });

      if ([0, 1].includes(alert.threshold) && ['VOLUME', 'VOLUME_POSITIVE', 'VOLUME_NEGATIVE'].includes(alert.type)) {
        return {
          VOLUME: 'En ' + interval + ' se ha producido <b>una nueva menci\xF3n</b> ' + category + ' a <b>' + feed.name + '</b> ' + source,
          VOLUME_POSITIVE: 'En ' + interval + ' se ha producido <b>una nueva menci\xF3n positiva</b> ' + category + ' a <b>' + feed.name + '</b> ' + source,
          VOLUME_NEGATIVE: 'En ' + interval + ' se ha producido <b>una nueva menci\xF3n negativa</b> ' + category + ' a <b>' + feed.name + '</b> ' + source
        }[alert.type];
      }

      return _fp2.default.getOr('Uknown alert type', alert.type, {
        VOLUME: 'En ' + interval + ' se han producido m\xE1s de <b>' + threshold + ' menciones</b> ' + category + ' a <b>' + feed.name + '</b> ' + source,
        VOLUME_POSITIVE: 'En ' + interval + ' se han producido m\xE1s de <b>' + threshold + ' menciones positivas</b> ' + category + ' a <b>' + feed.name + '</b> ' + source,
        VOLUME_NEGATIVE: 'En ' + interval + ' se han producido m\xE1s de <b>' + threshold + ' menciones negativas</b> ' + category + ' a <b>' + feed.name + '</b> ' + source,

        AUTHOR: '<b>' + atUsername + msg.username + '</b> acaba de mencionar ' + source + ' a <b>' + feed.name + '</b> ' + category,
        AUTHOR_NEGATIVE: '<b>' + atUsername + msg.username + '</b> acaba de mencionar ' + source + ' a <b>' + feed.name + '</b> negativamente ' + category,
        AUTHOR_POSITIVE: '<b>' + atUsername + msg.username + '</b> acaba de mencionar ' + source + ' a <b>' + feed.name + '</b> positivamente ' + category,

        INFLUENCE: 'Un usuario con m\xE1s de <b>' + threshold + ' seguidores</b> acaba de mencionar a <b>' + feed.name + '</b> ' + category + ' ' + source,
        INFLUENCE_NEGATIVE: 'Un usuario con m\xE1s de <b>' + threshold + ' seguidores</b> acaba de mencionar negativamente a <b>' + feed.name + '</b> ' + category + ' ' + source,
        INFLUENCE_POSITIVE: 'Un usuario con m\xE1s de <b>' + threshold + ' seguidores</b> acaba de mencionar positivamente a <b>' + feed.name + '</b> ' + category + ' ' + source
      });
    }
  }]);

  return AlertHeader;
}(_react2.default.PureComponent);

AlertHeader.contextTypes = {
  alert: _propTypes2.default.object,
  feed: _propTypes2.default.object,
  product: _propTypes2.default.object,
  messages: _propTypes2.default.array,
  mentions_filters: _propTypes2.default.object
};
exports.default = AlertHeader;
