// react-emails/lib/elements/Categories.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _urlFilters = require('@sentisis/url-filters');

var _Grid = require('../elements/Grid');

var _Grid2 = _interopRequireDefault(_Grid);

var _Title = require('../elements/Title');

var _Title2 = _interopRequireDefault(_Title);

var _Table = require('../elements/Table');

var _Table2 = _interopRequireDefault(_Table);

var _Separator = require('../elements/Separator');

var _Separator2 = _interopRequireDefault(_Separator);

var _Link = require('../elements/Link');

var _Link2 = _interopRequireDefault(_Link);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var styles = {
  Link: {
    margin: '35px 0 10px 0',
    textAlign: 'center'
  }
};

var keyValueShape = _propTypes2.default.shape({
  key: _propTypes2.default.string,
  value: _propTypes2.default.number
});

var Categories = function (_React$PureComponent) {
  _inherits(Categories, _React$PureComponent);

  function Categories() {
    _classCallCheck(this, Categories);

    return _possibleConstructorReturn(this, (Categories.__proto__ || Object.getPrototypeOf(Categories)).apply(this, arguments));
  }

  _createClass(Categories, [{
    key: 'render',
    value: function render() {
      var _context = this.context,
          _context$categories = _context.categories,
          categories = _context$categories === undefined ? [] : _context$categories,
          _context$concepts = _context.concepts,
          concepts = _context$concepts === undefined ? [] : _context$concepts,
          alert = _context.alert;

      var isAllSource = alert.source === 'all' ? 'en Twitter ' : '';
      if (!categories.length && !concepts.length) {
        return null;
      }
      var title = 'Las categor\xEDas y conceptos clave m\xE1s relevantes ' + isAllSource + 'est\xE1n siendo:';

      return _react2.default.createElement(
        _Grid2.default,
        null,
        _react2.default.createElement(_Separator2.default, null),
        _react2.default.createElement(
          _Title2.default,
          null,
          title
        ),
        _react2.default.createElement(
          _Grid2.default,
          null,
          _react2.default.createElement(
            _Grid2.default.Row,
            { className: 'Categories__table-container' },
            _react2.default.createElement(_Table2.default, { title: 'Categor\xEDas:', data: categories }),
            _react2.default.createElement(_Table2.default, { title: 'Conceptos:', data: concepts })
          )
        ),
        _react2.default.createElement(
          _Grid2.default,
          { style: styles.Link },
          _react2.default.createElement(
            _Link2.default,
            { button: true, to: this.buttonHref, className: 'AlertLayout__button' },
            'Sigue investigando en S\xE9ntisis'
          )
        )
      );
    }
  }, {
    key: 'buttonHref',
    get: function get() {
      var _context2 = this.context,
          brand = _context2.brand,
          product = _context2.product,
          mentions_filters = _context2.mentions_filters;

      var query = (0, _urlFilters.encode)(mentions_filters);
      return 'https://sentisis.io/' + brand.slug + '/' + product.slug + '/mentions/q=' + query + '?utm_source=alert&utm_medium=email';
    }
  }]);

  return Categories;
}(_react2.default.PureComponent);

Categories.contextTypes = {
  brand: _propTypes2.default.object,
  alert: _propTypes2.default.object,
  product: _propTypes2.default.object,
  mentions_filters: _propTypes2.default.object,

  categories: _propTypes2.default.arrayOf(keyValueShape),
  concepts: _propTypes2.default.arrayOf(keyValueShape)
};


Categories.propTypes = {
  categories: _propTypes2.default.array
};

exports.default = Categories;
