// react-emails/lib/elements/EvolutionChart.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _fp = require('lodash/fp');

var _fp2 = _interopRequireDefault(_fp);

var _momentTimezone = require('moment-timezone');

var _momentTimezone2 = _interopRequireDefault(_momentTimezone);

var _d3Shape = require('d3-shape');

var _d3Scale = require('d3-scale');

var _Grid = require('./Grid');

var _Grid2 = _interopRequireDefault(_Grid);

var _Title = require('./Title');

var _Title2 = _interopRequireDefault(_Title);

var _format = require('../utils/format');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var HEIGHT = 100;

var dataShape = _propTypes2.default.shape({
  timestamp: _propTypes2.default.number.isRequired,
  sentiment: _propTypes2.default.objectOf(_propTypes2.default.number).isRequired
});

var style = {

  wrapper: {
    marginBottom: '40px'
  },

  wrapperTable: {
    width: '100%',
    borderCollapse: 'collapse'
  },

  label: {
    fontSize: '12px',
    color: '#CCD1D8',
    textAlign: 'center',
    paddingTop: '5px'
  },

  title: {
    marginBottom: '20px'
  },

  yLabelsTable: {
    height: HEIGHT + 'px',
    fontSize: '12px',
    color: '#CCD1D8'
  },

  yLabelBottomCell: {
    verticalAlign: 'bottom'
  },

  yLabelTopCell: {
    verticalAlign: 'top'
  },

  legend: {
    marginRight: '15px',
    fontSize: '14px',
    color: '#9BA7B5'
  },

  legendIcon: {
    marginRight: '5px',
    paddingLeft: '12px'
  },

  legendContainer: {
    paddingBottom: '20px',
    textAlign: 'right'
  }

};

var EvolutionChart = function (_React$PureComponent) {
  _inherits(EvolutionChart, _React$PureComponent);

  function EvolutionChart() {
    _classCallCheck(this, EvolutionChart);

    return _possibleConstructorReturn(this, (EvolutionChart.__proto__ || Object.getPrototypeOf(EvolutionChart)).apply(this, arguments));
  }

  _createClass(EvolutionChart, [{
    key: 'render',
    value: function render() {
      var data = this.props.data;
      var alert = this.context.alert;


      var labels = [{
        key: 'OBJECTIVE',
        color: '#FFAA00',
        text: 'Neutro'
      }, {
        key: 'NEGATIVE',
        color: '#FF5A55',
        text: 'Negativo'
      }, {
        key: 'POSITIVE',
        color: '#79EAD0',
        text: 'Positivo'
      }];

      if (!data || data.length === 0) return null;

      var maxVal = _fp2.default.maxBy('count', data).count;
      var keys = labels.map(_fp2.default.get('key'));
      var createStack = (0, _d3Shape.stack)().keys(keys).value(function (d, key) {
        return _fp2.default.getOr(0, ['sentiment', key], d);
      }).order(_d3Shape.stackOrderDescending);

      var series = createStack(data);
      var y = (0, _d3Scale.scaleLinear)().domain([0, maxVal]).range([HEIGHT, 0]);

      var heights = [];
      var bars = series.map(function (bars, index) {
        // Get the colors from the labels
        var key = keys[index];
        var color = _fp2.default.getOr('#ff0000', 'color', _fp2.default.find({ key: key }, labels));

        return bars.map(function (d, i) {
          var height = y(d[0]) - y(d[1]);
          heights = _fp2.default.set([index, i], height, heights);

          // Dont' render the bar if it's less than 1px in height
          if (height < 1) return null;

          var style = {
            margin: 0,
            lineHeight: 0,
            borderTop: height + 'px solid ' + color
          };

          return _react2.default.createElement(
            'p',
            { key: '' + key + i, style: style },
            '\xA0'
          );
        });
      });

      var $columns = data.map(function (d, index) {
        var elements = bars.map(function (b) {
          return b[index];
        });

        // The paddingTop is used at this point to put the clearence between the
        // end of the bar and the chart. It's the remaining space when subtracting
        // to the height of the chart the values of the bars.
        var top = heights.reduce(function (acc, v) {
          return acc + v[index];
        }, 0);

        var style = {
          paddingTop: HEIGHT - top + 'px',
          paddingLeft: '2px',
          paddingRight: '2px'
        };

        return _react2.default.createElement(
          'td',
          { style: style, key: index },
          elements
        );
      });

      var labelFormat = alert.interval === 'D' ? 'DD/MM' : 'HH:mm';
      var $xLabels = data.map(function (d, index) {
        return _react2.default.createElement(
          'td',
          { style: style.label, key: index },
          _momentTimezone2.default.tz(d.timestamp, alert.timezone || 'Europe/Madrid').format(labelFormat)
        );
      });

      var $yLabels = _react2.default.createElement(
        _Grid2.default,
        { style: style.yLabelsTable },
        _react2.default.createElement(
          _Grid2.default.Cell,
          { style: style.yLabelTopCell },
          (0, _format.formatNumber)(maxVal, { decimals: 0 })
        ),
        _react2.default.createElement(
          _Grid2.default.Cell,
          { style: style.yLabelBottomCell },
          '0'
        )
      );

      var $legends = labels.map(function (label) {
        return _react2.default.createElement(
          'span',
          { key: label.key, style: style.legend },
          _react2.default.createElement(
            'span',
            { style: Object.assign({}, style.legendIcon, { backgroundColor: label.color }) },
            '\xA0'
          ),
          label.text
        );
      });

      return _react2.default.createElement(
        _Grid2.default,
        { style: style.wrapper },
        _react2.default.createElement(
          _Title2.default,
          { style: style.title },
          this.title
        ),
        _react2.default.createElement(
          _Grid2.default.Cell,
          { style: style.legendContainer },
          $legends
        ),
        _react2.default.createElement(
          'table',
          { style: style.wrapperTable },
          _react2.default.createElement(
            'tbody',
            null,
            _react2.default.createElement(
              'tr',
              null,
              _react2.default.createElement(
                'td',
                null,
                $yLabels
              ),
              $columns
            ),
            _react2.default.createElement(
              'tr',
              null,
              _react2.default.createElement('td', null),
              $xLabels
            )
          )
        )
      );
    }
  }, {
    key: 'title',
    get: function get() {
      var alert = this.context.alert;

      var category = alert.category_name ? 'sobre ' + alert.category_name : '';
      var conversation = _fp2.default.getOr('', alert.type, {
        VOLUME: 'conversación',
        VOLUME_POSITIVE: 'conversación positiva',
        VOLUME_NEGATIVE: 'conversación negativa'
      });
      var interval = _fp2.default.getOr('', alert.interval, {
        M: 'en los últimos 10 minutos',
        H: 'en las últimas 6 horas:',
        D: 'en los últimos 7 días:'
      });
      return 'As\xED ha evolucionado la ' + conversation + ' ' + category + ' ' + interval;
    }
  }]);

  return EvolutionChart;
}(_react2.default.PureComponent);

EvolutionChart.contextTypes = {
  alert: _propTypes2.default.object.isRequired
};


EvolutionChart.propTypes = {
  data: _propTypes2.default.arrayOf(dataShape).isRequired
};

exports.default = EvolutionChart;
