// react-emails/lib/elements/Footer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _fp = require('lodash/fp');

var _fp2 = _interopRequireDefault(_fp);

var _Link = require('./Link');

var _Link2 = _interopRequireDefault(_Link);

var _Img = require('./Img');

var _Img2 = _interopRequireDefault(_Img);

var _Grid = require('./Grid');

var _Grid2 = _interopRequireDefault(_Grid);

var _locale = require('../utils/locale');

var _locale2 = _interopRequireDefault(_locale);

var _types = require('../types');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var logoSrc = 'https://s3-eu-west-1.amazonaws.com/sentisis-images/emails/logo.png';
var footerSrc = 'https://s3-eu-west-1.amazonaws.com/sentisis-images/emails/footer.png';
var appHref = 'https://sentisis.io';
var learningCenterHref = 'http://learning.sentisis.io';

var styles = {

  container: {
    marginTop: '25px',
    width: '100%'
  },

  textCol: {
    fontSize: '13px',
    lineHeight: '140%',
    color: '#A4AFBB',
    paddingRight: '35px'
  },

  logoCol: {},

  p: {
    margin: '0'
  },

  links: {
    fontSize: '12px',
    textDecoration: 'none',
    fontWeight: 'bold',
    color: '#7F9BB5',
    border: 'none'
  },

  copyright: {
    fontSize: '12px',
    color: '#7F9BB5',
    lineHeight: '170%'
  },

  copyrightFooter: {
    display: 'block',
    margin: '25px 30px 20px 30px'
  },

  footerImage: {
    width: '100%'
  }

};

var Footer = function (_React$Component) {
  _inherits(Footer, _React$Component);

  function Footer() {
    _classCallCheck(this, Footer);

    return _possibleConstructorReturn(this, (Footer.__proto__ || Object.getPrototypeOf(Footer)).apply(this, arguments));
  }

  _createClass(Footer, [{
    key: 'render',
    value: function render() {
      var type = this.props.type;


      switch (type) {
        case _types.PASSWORD_RESET:
        case _types.USER_CREATED:
          return _react2.default.createElement(
            _Grid2.default,
            null,
            _react2.default.createElement(
              _Grid2.default,
              { style: styles.copyrightFooter },
              _react2.default.createElement(
                _Grid2.default.Row,
                null,
                _react2.default.createElement(
                  'p',
                  null,
                  _react2.default.createElement(
                    _Link2.default,
                    { simple: true, to: appHref },
                    'S\xE9ntisis.io |'
                  ),
                  ' ',
                  _react2.default.createElement(
                    _Link2.default,
                    { simple: true, to: learningCenterHref },
                    'Learning Center'
                  )
                ),
                _react2.default.createElement(
                  'p',
                  { style: styles.copyright },
                  'Copyright \xA9 S\xE9ntisis Analytics 2017. Todos los derechos reservados. Puedes encontrarnos en Calle Gran V\xEDa 62, 6\xBAIzq. Madrid 28004, Espa\xF1a.'
                )
              )
            ),
            _react2.default.createElement(_Img2.default, { className: 'Footer-border__img', alt: 'Footer', src: footerSrc })
          );
        default:
          return _react2.default.createElement(
            _Grid2.default,
            { style: styles.container },
            _react2.default.createElement(
              _Grid2.default.Row,
              null,
              _react2.default.createElement(
                _Grid2.default.Cell,
                { style: styles.textCol, className: 'Footer__left' },
                this.footerCopy
              ),
              _react2.default.createElement(
                _Grid2.default.Cell,
                { style: styles.logoCol, className: 'Footer__right' },
                _react2.default.createElement(_Img2.default, { className: 'Footer__img', alt: 'S\xE9ntisis logo', src: logoSrc })
              )
            )
          );
      }
    }
  }, {
    key: 'configHref',
    get: function get() {
      var type = this.props.type;
      var _context = this.context,
          brand = _context.brand,
          product = _context.product;


      var base = 'https://sentisis.io/' + brand.slug + '/' + product.slug;

      switch (type) {

        case _types.ALERT:
          return base + '/alerts';

        case _types.DOWNLOAD_MENTIONS:
          return 'https://sentisis.io/profile';

        default:
          return base + '/config';

      }
    }
  }, {
    key: 'footerCopy',
    get: function get() {
      var type = this.props.type;
      var _context2 = this.context,
          product = _context2.product,
          alert = _context2.alert,
          download = _context2.download;


      var productString = _fp2.default.capitalize(product.type) + ': ' + product.name;
      var productDownloadString = _fp2.default.capitalize(download.productType) + ': ' + download.productName;

      switch (type) {

        case _types.ALERT:
          return _react2.default.createElement(
            'p',
            { style: styles.p },
            'Has recibido este email por la',
            _react2.default.createElement(
              'strong',
              null,
              ' alerta por ',
              (0, _locale2.default)('alert', alert.type),
              ': ',
              alert.name
            ),
            ', en tu producto ',
            _react2.default.createElement(
              'strong',
              null,
              productString
            ),
            '. Puedes modificar la configuraci\xF3n de tus alertas ',
            _react2.default.createElement(
              _Link2.default,
              { to: this.configHref },
              'aqu\xED'
            )
          );

        case _types.DOWNLOAD_MENTIONS:
          return _react2.default.createElement(
            'p',
            { style: styles.p },
            'Has recibido este email por una solicitud de descarga en tu producto ',
            _react2.default.createElement(
              'strong',
              null,
              productDownloadString
            ),
            '. Puedes modificar tu email de contacto ',
            _react2.default.createElement(
              _Link2.default,
              { to: this.configHref },
              'aqu\xED'
            ),
            '.'
          );

        default:
          return _react2.default.createElement(
            'p',
            { style: styles.p },
            'Puedes modificar tu email de contacto ',
            _react2.default.createElement(
              _Link2.default,
              { to: this.configHref },
              'aqu\xED'
            ),
            '.'
          );

      }
    }
  }]);

  return Footer;
}(_react2.default.Component);

Footer.propTypes = {
  type: _propTypes2.default.string.isRequired
};
Footer.contextTypes = {
  brand: _propTypes2.default.object,
  product: _propTypes2.default.object,
  alert: _propTypes2.default.object,
  download: _propTypes2.default.object
};
exports.default = Footer;
