// react-emails/lib/elements/Img.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  alt: _propTypes2.default.string.isRequired,
  src: _propTypes2.default.string.isRequired
};

var styles = {

  img: {
    outline: 'none',
    textDecoration: 'none',
    border: 'none',
    display: 'block'
  }

};

function Img(_ref) {
  var src = _ref.src,
      alt = _ref.alt,
      className = _ref.className,
      others = _objectWithoutProperties(_ref, ['src', 'alt', 'className']);

  return _react2.default.createElement('img', Object.assign({ src: src, alt: alt, style: styles.img, className: className }, others));
}

Img.propTypes = propTypes;

exports.default = Img;
