// react-emails/lib/elements/Link.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propTypes = {
  to: _propTypes2.default.string.isRequired,

  children: _propTypes2.default.node.isRequired,

  button: _propTypes2.default.bool,

  // Shows a text link intead of a button
  simple: _propTypes2.default.bool,

  blue: _propTypes2.default.bool
};

var styles = {

  link: {
    color: '#10D4AB',
    fontWeight: 'bold'
  },

  button: {
    backgroundColor: '#10D4AB',
    textDecoration: 'none',
    color: 'white',
    border: '15px solid #10D4AB',
    marginRight: '15px',
    lineHeight: '42px',
    textTransform: 'uppercase',
    fontSize: '13px',
    fontWeight: 'bold',
    letterSpacing: '0.5px',
    boxShadow: '#16BA98 0px 5px 0px 0px'
  },

  simple: {
    color: '#7F9BB5',
    fontSize: '12px',
    fontWeight: 'bold'
  }

};

function Link(_ref) {
  var to = _ref.to,
      children = _ref.children,
      button = _ref.button,
      simple = _ref.simple,
      blue = _ref.blue,
      className = _ref.className;

  var style = button ? styles.button : styles.link;
  if (simple) style = styles.simple;

  if (blue) {
    style = Object.assign({}, style, {
      color: '#3AA9E3'
    });
  }

  return _react2.default.createElement(
    'a',
    { style: style, href: to, className: className },
    children
  );
}

Link.propTypes = propTypes;
exports.default = Link;
