// react-emails/lib/elements/Message.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _momentTimezone = require('moment-timezone');

var _momentTimezone2 = _interopRequireDefault(_momentTimezone);

var _format = require('../utils/format');

var _Grid = require('./Grid');

var _Grid2 = _interopRequireDefault(_Grid);

var _Link = require('./Link');

var _Link2 = _interopRequireDefault(_Link);

var _fp = require('lodash/fp');

var _fp2 = _interopRequireDefault(_fp);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propTypes = {
  name: _propTypes2.default.string,
  username: _propTypes2.default.string,
  content: _propTypes2.default.string,
  source: _propTypes2.default.string,
  followers: _propTypes2.default.any
};

var styles = {

  wrapper: {
    border: '2px solid #F0F2F7',
    marginBottom: '20px',
    marginTop: '20px'
  },

  container: {
    margin: '20px',
    width: undefined
  },

  date: {
    fontSize: '14px',
    display: 'inline-block'
  },

  name: {
    fontSize: '14px',
    fontWeight: '600',
    color: '#4A637C',
    paddingRight: '10px'
  },

  username: {
    fontSize: '14px',
    fontWeight: '600',
    color: '#A4AFBA',
    paddingRight: '10px'
  },

  followers: {
    display: 'inline-block',
    fontSize: '12px',
    fontWeight: '600',
    marginRight: '10px',
    borderTop: '2px solid #e0e3e8',
    borderBottom: '2px solid #e0e3e8',
    borderLeft: '4px solid #e0e3e8',
    borderRight: '4px solid #e0e3e8',
    color: '#7b8d9e',
    borderRadius: '2px',
    backgroundColor: '#e0e3e8'
  },

  content: {
    fontSize: '14px',
    fontWeight: '500',
    color: '#4A637C',
    lineHeight: '140%',
    marginTop: '10px',
    marginBottom: 0
  },

  stats: {
    container: {
      display: 'block',
      marginTop: '10px'
    },
    block: {
      marginRight: '10px'
    },
    label: {
      color: '#A4AFBA',
      textTransform: 'uppercase',
      fontSize: '12px'
    },
    value: {
      color: '#4A637C',
      fontSize: '12px'
    }
  }

};

function Message(props) {
  var source = props.source,
      date = props.date,
      timezone = props.timezone,
      link = props.link;

  var dateFormat = 'DD/MM/YY - HH:mm';
  var $text = 'La mención no está disponible actualmente';
  if (props.text) {
    $text = (0, _format.shortenText)(props.text, 350);
  }

  var $name = props.name && _react2.default.createElement(
    'span',
    { style: styles.name, className: 'Message__name' },
    (0, _format.shortenText)(props.name, 50)
  );
  var $date = date && _react2.default.createElement(
    'span',
    { style: styles.date, className: 'Message__date' },
    _react2.default.createElement(
      _Link2.default,
      { simple: true, to: link },
      _momentTimezone2.default.tz(date, timezone).format(dateFormat)
    )
  );

  var $username = void 0;
  var $followers = void 0;
  if (source === 'tw' || source === 'ig') {
    $username = props.username && _react2.default.createElement(
      'span',
      { style: styles.username, className: 'Message__username' },
      '@',
      props.username
    );
    $followers = props.followers && _react2.default.createElement(
      'div',
      { style: styles.followers, className: 'Message__followers' },
      (0, _format.formatNumber)(props.followers, { shorten: false }),
      ' Seguidores'
    );
  }

  // Message stats
  var validStats = ['retweets', 'likes', 'reach', 'views', 'comments'];
  var stats = validStats.filter(function (key) {
    return _fp2.default.includes(key, Object.keys(props)) && props[key];
  });
  var mapStatName = function mapStatName(fieldName) {
    return {
      comments: 'Comentarios',
      reach: 'Alcance'
    }[fieldName] || fieldName;
  };

  var $stats = void 0;
  if (stats) {
    $stats = _react2.default.createElement(
      _Grid2.default.Cell,
      { className: 'Message__stats-container', style: styles.stats.container },
      stats.map(function (stat) {
        var label = _react2.default.createElement(
          'span',
          { style: styles.stats.label },
          mapStatName(stat)
        );
        var value = _react2.default.createElement(
          'span',
          { style: styles.stats.value },
          (0, _format.formatNumber)(props[stat])
        );

        return _react2.default.createElement(
          'span',
          { className: 'Message__stat', key: stat, style: styles.stats.block },
          value,
          ' ',
          label
        );
      })
    );
  }

  return _react2.default.createElement(
    _Grid2.default,
    { style: styles.wrapper },
    _react2.default.createElement(
      _Grid2.default,
      { style: styles.container },
      _react2.default.createElement(
        _Grid2.default.Cell,
        null,
        $name,
        $username,
        $followers,
        $date
      ),
      _react2.default.createElement(
        'p',
        { style: styles.content },
        $text
      ),
      $stats
    )
  );
}

Message.propTypes = propTypes;
exports.default = Message;
