// react-emails/lib/elements/MessageContainer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _format = require('../utils/format');

var _Message = require('./Message');

var _Message2 = _interopRequireDefault(_Message);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var MessageContainer = function (_React$PureComponent) {
  _inherits(MessageContainer, _React$PureComponent);

  function MessageContainer() {
    _classCallCheck(this, MessageContainer);

    return _possibleConstructorReturn(this, (MessageContainer.__proto__ || Object.getPrototypeOf(MessageContainer)).apply(this, arguments));
  }

  _createClass(MessageContainer, [{
    key: 'render',
    value: function render() {
      var source = this.props.source;
      var timezone = this.context.alert.timezone;

      var message = this.props.data;
      var name = void 0;
      var username = void 0;
      var followers = void 0;

      // Set name, username, photo and followers
      if (message.name) name = message.name;
      if (message.username) username = message.username;
      if (message.followers !== undefined) followers = (0, _format.formatNumber)(message.followers);

      if (source === 'facebook' && !name) {
        name = username;
      }

      return _react2.default.createElement(_Message2.default, Object.assign({
        name: name,
        username: username,
        source: source,
        followers: followers,
        timezone: timezone
      }, message));
    }
  }]);

  return MessageContainer;
}(_react2.default.PureComponent);

MessageContainer.propTypes = {
  data: _propTypes2.default.object.isRequired
};
MessageContainer.contextTypes = {
  alert: _propTypes2.default.object.isRequired
};
exports.default = MessageContainer;
