// react-emails/lib/elements/MessagesList.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _fp = require('lodash/fp');

var _fp2 = _interopRequireDefault(_fp);

var _urlFilters = require('@sentisis/url-filters');

var _Title = require('./Title');

var _Title2 = _interopRequireDefault(_Title);

var _Link = require('../elements/Link');

var _Link2 = _interopRequireDefault(_Link);

var _MessageContainer = require('./MessageContainer');

var _MessageContainer2 = _interopRequireDefault(_MessageContainer);

var _Grid = require('./Grid');

var _Grid2 = _interopRequireDefault(_Grid);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var styles = {
  Link: {
    'margin': '30px 0 15px 0',
    'textAlign': 'right'
  },
  LinkCentered: {
    'margin': '30px 0 15px 0',
    'textAlign': 'center'
  }
};

var MessagesList = function (_React$PureComponent) {
  _inherits(MessagesList, _React$PureComponent);

  function MessagesList() {
    _classCallCheck(this, MessagesList);

    return _possibleConstructorReturn(this, (MessagesList.__proto__ || Object.getPrototypeOf(MessagesList)).apply(this, arguments));
  }

  _createClass(MessagesList, [{
    key: 'getAtUsername',
    value: function getAtUsername(message) {
      return message && ['tw', 'ig'].includes(message.source) ? '@' : '';
    }
  }, {
    key: 'renderTriggeredMessageBlock',
    value: function renderTriggeredMessageBlock() {
      var message = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};

      var fromUser = '';
      if (message.username) {
        fromUser = 'de ' + this.getAtUsername(message) + message.username;
      }

      var title = 'Esta menci\xF3n ' + fromUser + ' ha activado la alerta:';

      return _react2.default.createElement(
        'span',
        null,
        _react2.default.createElement(
          _Title2.default,
          null,
          title
        ),
        _react2.default.createElement(_MessageContainer2.default, { data: message, source: message.source })
      );
    }
  }, {
    key: 'renderContextMessagesBlock',
    value: function renderContextMessagesBlock(messages) {
      // There are no context messages, only the trigger message
      if (messages.length === 0) return null;

      var _context = this.context,
          alert = _context.alert,
          brand = _context.brand;

      var source = messages[0].source;
      var isVolume = alert.type.includes('VOLUME');
      var msgList = messages.map(function (msg, index) {
        return _react2.default.createElement(_MessageContainer2.default, { key: index, data: msg, source: msg.source });
      });
      var title = messages.length === 1 ? 'Esta es la última mención de la alerta:' : 'Estas son las últimas menciones de la alerta:';

      // Show a different title for the sources that have reach
      if (['tw', 'ig', 'yt'].includes(source)) {
        var isTwitter = source === 'tw' ? 'en Twitter' : '';
        title = messages.length === 1 ? 'Esta es la menci\xF3n con m\xE1s impacto ' + isTwitter + ' de la alerta:' : 'Estas son las menciones con m\xE1s impacto ' + isTwitter + ' de la alerta:';
      }

      // Author alerts
      if (!isVolume) {
        title = '' + this.getAtUsername(messages[0]) + messages[0].username + ' ya hab\xEDa mencionado ' + (messages.length === 1 ? 'una vez a ' + brand.name + ':' : 'a ' + brand.name + ' antes:');
      }

      return _react2.default.createElement(
        'span',
        null,
        _react2.default.createElement(
          _Title2.default,
          null,
          title
        ),
        msgList
      );
    }
  }, {
    key: 'render',
    value: function render() {
      var _context$messages = this.context.messages,
          messages = _context$messages === undefined ? [] : _context$messages;
      var alert = this.context.alert;

      var isVolume = alert.type.includes('VOLUME');
      messages = isVolume ? _fp2.default.sortBy('reach', messages).reverse() : _fp2.default.sortBy('date', messages).reverse();
      var triggeredMessage = _fp2.default.head(messages);

      // Author/Influence: Shows other messages than the one that triggered the alert
      var contextMessages = _fp2.default.tail(messages);
      var $messagesList = this.renderContextMessagesBlock(contextMessages);
      var footer = _react2.default.createElement(
        _Grid2.default,
        { style: styles.LinkCentered },
        _react2.default.createElement(
          _Grid2.default.Cell,
          null,
          _react2.default.createElement(
            _Link2.default,
            { className: 'AlertLayout__button', button: true, to: this.buttonHref },
            'Ver la actividad completa en S\xE9ntisis'
          )
        )
      );

      // Volume:
      // The message list shows all the messages, including the first one
      // The link is not a button but a text link
      if (isVolume) {
        $messagesList = this.renderContextMessagesBlock(messages);
        footer = _react2.default.createElement(
          _Grid2.default,
          { style: styles.Link },
          _react2.default.createElement(
            _Grid2.default.Cell,
            null,
            _react2.default.createElement(
              _Link2.default,
              { className: 'MessagesList__link', to: this.buttonHref },
              'Ver todas las menciones en S\xE9ntisis'
            )
          )
        );
      }

      // Volume alerts don't show the triggered mention
      return _react2.default.createElement(
        'span',
        null,
        !isVolume && this.renderTriggeredMessageBlock(triggeredMessage),
        $messagesList,
        footer
      );
    }
  }, {
    key: 'buttonHref',
    get: function get() {
      var _context2 = this.context,
          brand = _context2.brand,
          product = _context2.product,
          mentions_filters = _context2.mentions_filters;

      var query = (0, _urlFilters.encode)(mentions_filters);
      return 'https://sentisis.io/' + brand.slug + '/' + product.slug + '/mentions/q=' + query + '?utm_source=alert&utm_medium=email';
    }
  }]);

  return MessagesList;
}(_react2.default.PureComponent);

MessagesList.contextTypes = {
  alert: _propTypes2.default.object,
  messages: _propTypes2.default.array,
  brand: _propTypes2.default.object,
  mentions_filters: _propTypes2.default.object,
  product: _propTypes2.default.object
};
exports.default = MessagesList;
