// react-emails/lib/elements/Table.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Grid = require('../elements/Grid');

var _Grid2 = _interopRequireDefault(_Grid);

var _Title = require('../elements/Title');

var _Title2 = _interopRequireDefault(_Title);

var _fp = require('lodash/fp');

var _fp2 = _interopRequireDefault(_fp);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = {
  subtitle: {
    fontSize: '14px',
    margin: '20px 0'
  },
  table: {
    width: '270px',
    marginRight: '20px',
    borderSpacing: '0px',
    fontSize: '13px'
  },
  tableCell: {
    padding: '8px 15px'
  }
};

var getRowStyles = function getRowStyles(index) {
  return {
    height: '15px',
    backgroundColor: index % 2 === 0 ? '#EEE' : '#FFF'
  };
};

var Table = function Table(_ref) {
  var title = _ref.title,
      data = _ref.data;
  return _react2.default.createElement(
    _Grid2.default,
    null,
    _react2.default.createElement(
      _Grid2.default.Cell,
      null,
      _react2.default.createElement(
        _Title2.default,
        { style: styles.subtitle },
        title
      ),
      _react2.default.createElement(
        'table',
        { className: 'Table', style: styles.table },
        _react2.default.createElement(
          'tbody',
          null,
          _fp2.default.filter(function (r) {
            return r.key && r.value;
          }, data).map(function (r, index) {
            return _react2.default.createElement(
              'tr',
              { className: 'row', style: getRowStyles(index), key: r.key },
              _react2.default.createElement(
                'td',
                { style: styles.tableCell },
                r.key
              ),
              _react2.default.createElement(
                'td',
                { style: styles.tableCell },
                r.value
              )
            );
          })
        )
      )
    )
  );
};

Table.propTypes = {
  title: _propTypes2.default.string.isRequired,
  data: _propTypes2.default.array.isRequired
};

exports.default = Table;
