// react-emails/lib/elements/Wrapper.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Footer = require('./Footer');

var _Footer2 = _interopRequireDefault(_Footer);

var _Header = require('./Header');

var _Header2 = _interopRequireDefault(_Header);

var _AlertHeader = require('./AlertHeader');

var _AlertHeader2 = _interopRequireDefault(_AlertHeader);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propTypes = {
  children: _propTypes2.default.node.isRequired,
  type: _propTypes2.default.string.isRequired
};

var styles = {

  table: {
    margin: 0,
    padding: 0,
    width: '100%',
    lineHeight: '100%',
    backgroundColor: '#F2F4F7',
    border: 0,
    fontFamily: 'Arial, Helvetica, sans-serif',
    fontSize: '14px',
    color: '#4A637C'
  },

  innnerTable: {
    maxWidth: '670px',
    width: '100%',
    margin: '30px auto',
    padding: '0 20px'
  },

  body: {
    backgroundColor: 'white',
    padding: '25px',
    borderRadius: '5px'
  },

  wrapperTableContainer: {
    padding: '0 5px'
  },

  wrapperTable: {
    width: '100%'
  }

};

function Wrapper(_ref) {
  var children = _ref.children,
      type = _ref.type;

  var header = void 0;
  var $alertHeader = void 0;

  // The super-alerts have the header outside the wrapper
  if (type === 'alert') {
    $alertHeader = _react2.default.createElement(_AlertHeader2.default, null);
  }

  // Wrapper overrides when creating a user/resetting a password
  if (['passwordReset', 'userCreated'].includes(type)) {
    styles.table = Object.assign({}, styles.table, {
      backgroundColor: 'white'
    });

    styles.innnerTable = Object.assign({}, styles.innnerTable, {
      paddingBottom: '30px'

      // Add global border to the e-mail content
    });styles.body = Object.assign({}, styles.body, {
      display: 'block',
      marginTop: '30px',
      border: '3px solid #F1F6F8',
      padding: '40px 30px'
    });

    // These 2 layouts use a header
    header = _react2.default.createElement(_Header2.default, null);
  }

  return _react2.default.createElement(
    'table',
    { style: styles.table },
    _react2.default.createElement(
      'tbody',
      null,
      _react2.default.createElement(
        'tr',
        null,
        _react2.default.createElement(
          'td',
          { style: styles.wrapperTableContainer },
          _react2.default.createElement(
            'table',
            { style: styles.wrapperTable },
            _react2.default.createElement(
              'tbody',
              null,
              _react2.default.createElement(
                'tr',
                null,
                _react2.default.createElement(
                  'td',
                  null,
                  _react2.default.createElement(
                    'table',
                    { style: styles.innnerTable },
                    _react2.default.createElement(
                      'tbody',
                      null,
                      type === 'alert' ? _react2.default.createElement(
                        'tr',
                        null,
                        _react2.default.createElement(
                          'td',
                          null,
                          $alertHeader
                        )
                      ) : null,
                      _react2.default.createElement(
                        'tr',
                        null,
                        header
                      ),
                      _react2.default.createElement(
                        'tr',
                        null,
                        _react2.default.createElement(
                          'td',
                          { style: styles.body },
                          children
                        )
                      ),
                      _react2.default.createElement(
                        'tr',
                        null,
                        _react2.default.createElement(
                          'td',
                          null,
                          _react2.default.createElement(_Footer2.default, { type: type })
                        )
                      )
                    )
                  )
                )
              )
            )
          )
        )
      )
    )
  );
}

Wrapper.propTypes = propTypes;

exports.default = Wrapper;
