// react-emails/lib/utils/locale.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

exports.default = function (type, key) {
  return _fp2.default.getOr('', [type, key], {

    source: {
      fb: 'Facebook',
      tw: 'Twitter',
      ig: 'Instagram',
      yt: 'Youtube',
      media: 'Medios',
      all: 'todas las fuentes'
    },

    filters: {
      fb: 'facebook',
      tw: 'twitter',
      ig: 'instagram',
      yt: 'youtube',
      media: 'media'
    },

    media: {
      news: 'Noticia',
      forum: 'Foro',
      blog: 'Blog'
    },

    alert: {
      INFLUENCE: 'influencia',
      INFLUENCE_NEGATIVE: 'influencia negativa',
      INFLUENCE_POSITIVE: 'influencia positiva',
      VOLUME: 'volumen',
      VOLUME_POSITIVE: 'volumen positiva',
      VOLUME_NEGATIVE: 'volumen negativa',
      AUTHOR: 'autor',
      AUTHOR_POSITIVE: 'autor positiva',
      AUTHOR_NEGATIVE: 'autor negative'
    }

  });
};

var _fp = require('lodash/fp');

var _fp2 = _interopRequireDefault(_fp);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
