// react-emails/lib/layouts/AlertLayout.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _fp = require('lodash/fp');

var _fp2 = _interopRequireDefault(_fp);

var _Grid = require('../elements/Grid');

var _Grid2 = _interopRequireDefault(_Grid);

var _Img = require('../elements/Img');

var _Img2 = _interopRequireDefault(_Img);

var _MessagesList = require('../elements/MessagesList');

var _MessagesList2 = _interopRequireDefault(_MessagesList);

var _Categories = require('../elements/Categories');

var _Categories2 = _interopRequireDefault(_Categories);

var _EvolutionChart = require('../elements/EvolutionChart');

var _EvolutionChart2 = _interopRequireDefault(_EvolutionChart);

var _locale = require('../utils/locale');

var _locale2 = _interopRequireDefault(_locale);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var contextIcon = 'https://s3-eu-west-1.amazonaws.com/sentisis-images/emails/context-icon.png';

var styles = {
  icon: {
    verticalAlign: 'top',
    paddingRight: '10px'
  },
  alertTitle: {
    display: 'block',
    fontWeight: 'bold',
    paddingTop: '5px',
    fontSize: '13px'
  },
  header: {
    display: 'block',
    paddingBottom: '20px'
  }
};

var AlertLayout = function (_React$PureComponent) {
  _inherits(AlertLayout, _React$PureComponent);

  function AlertLayout() {
    _classCallCheck(this, AlertLayout);

    return _possibleConstructorReturn(this, (AlertLayout.__proto__ || Object.getPrototypeOf(AlertLayout)).apply(this, arguments));
  }

  _createClass(AlertLayout, [{
    key: 'render',
    value: function render() {
      var alert = this.context.alert;

      var isVolume = alert.type.includes('VOLUME');

      return _react2.default.createElement(
        _Grid2.default,
        null,
        this.contextHeader,
        isVolume ? _react2.default.createElement(_EvolutionChart2.default, { data: this.context.histogram }) : null,
        _react2.default.createElement(_MessagesList2.default, null),
        isVolume ? _react2.default.createElement(_Categories2.default, null) : null
      );
    }
  }, {
    key: 'contextHeader',
    get: function get() {
      return _react2.default.createElement(
        _Grid2.default,
        null,
        _react2.default.createElement(
          _Grid2.default.Cell,
          null,
          _react2.default.createElement(
            _Grid2.default,
            null,
            _react2.default.createElement(
              _Grid2.default.Row,
              { style: styles.header },
              _react2.default.createElement(
                _Grid2.default.Cell,
                { style: styles.icon },
                _react2.default.createElement(_Img2.default, { width: '24px', height: '24px', alt: 'Alert icon', src: contextIcon })
              ),
              _react2.default.createElement(
                _Grid2.default.Cell,
                null,
                _react2.default.createElement(
                  'span',
                  { style: styles.alertTitle },
                  'Contexto de la Alerta'
                )
              )
            )
          )
        )
      );
    }
  }]);

  return AlertLayout;
}(_react2.default.PureComponent);

AlertLayout.contextTypes = {
  alert: _propTypes2.default.object,
  feed: _propTypes2.default.object,
  product: _propTypes2.default.object,
  messages: _propTypes2.default.array,
  mentions_filters: _propTypes2.default.object,
  histogram: _propTypes2.default.arrayOf(_propTypes2.default.object)
};
exports.default = AlertLayout;
