// react-emails/lib/layouts/DownloadLayout.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _Title = require('../elements/Title');

var _Title2 = _interopRequireDefault(_Title);

var _Grid = require('../elements/Grid');

var _Grid2 = _interopRequireDefault(_Grid);

var _Link = require('../elements/Link');

var _Link2 = _interopRequireDefault(_Link);

var _format = require('../utils/format');

var _locale = require('../utils/locale');

var _locale2 = _interopRequireDefault(_locale);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var styles = {
  lastMentionBlock: {
    fontSize: '16px',
    color: '#4A637C',
    marginBottom: 0
  },

  title: {
    marginBottom: '30px'
  },

  expirationCopy: {
    color: '#10D4AB',
    fontSize: '14px',
    fontWeight: 'bold'
  }
};

var DownloadLayout = function (_React$Component) {
  _inherits(DownloadLayout, _React$Component);

  function DownloadLayout() {
    _classCallCheck(this, DownloadLayout);

    return _possibleConstructorReturn(this, (DownloadLayout.__proto__ || Object.getPrototypeOf(DownloadLayout)).apply(this, arguments));
  }

  _createClass(DownloadLayout, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        _Grid2.default,
        null,
        _react2.default.createElement(
          _Title2.default,
          { style: styles.title },
          this.title
        ),
        _react2.default.createElement(
          _Grid2.default.Cell,
          null,
          _react2.default.createElement(
            _Link2.default,
            { button: true, to: this.buttonHref, className: 'DownloadLayout__button' },
            'Descargar menciones'
          )
        ),
        this.expirationCopy
      );
    }
  }, {
    key: 'title',
    get: function get() {
      // const mentionsCount
      var download = this.context.download;

      var source = (0, _locale2.default)('source', download.source);
      var mentionsCount = (0, _format.formatNumber)(download.numMessages, { shorten: false });
      var startDate = (0, _moment2.default)(new Date(Number(download.startDate))).format('DD/MM/YY');
      var endDate = (0, _moment2.default)(new Date(Number(download.endDate))).format('DD/MM/YY');
      var period = 'desde el ' + startDate + ' hasta el ' + endDate;

      return 'Has solicitado descargar ' + mentionsCount + ' menciones de ' + source + ' ' + period + '.';
    }
  }, {
    key: 'buttonHref',
    get: function get() {
      var download = this.context.download;

      return download.downloadLink;
    }
  }, {
    key: 'expirationCopy',
    get: function get() {
      return _react2.default.createElement(
        _Grid2.default.Cell,
        null,
        _react2.default.createElement(
          'p',
          { style: styles.expirationCopy },
          'Este enlace estar\xE1 disponible durante 3 meses'
        )
      );
    }
  }]);

  return DownloadLayout;
}(_react2.default.Component);

DownloadLayout.contextTypes = {
  alert: _propTypes2.default.object,
  brand: _propTypes2.default.object,
  product: _propTypes2.default.object,
  message: _propTypes2.default.object,
  mentions_filters: _propTypes2.default.object,
  download: _propTypes2.default.object
};
exports.default = DownloadLayout;
