// react-emails/lib/layouts/LangTextLayout.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Grid = require('../elements/Grid');

var _Grid2 = _interopRequireDefault(_Grid);

var _Header = require('../lang/Header');

var _Header2 = _interopRequireDefault(_Header);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var style = {
  body: {
    lineHeight: '1.5',
    paddingBottom: '20px'
  }
};

function LangTextLayout(props, _ref) {
  var title = _ref.title,
      bodyPlainText = _ref.bodyPlainText,
      bodyHTML = _ref.bodyHTML;

  return _react2.default.createElement(
    _Grid2.default,
    null,
    _react2.default.createElement(_Header2.default, {
      name: 'Lang.ai',
      title: title
    }),
    bodyHTML ? _react2.default.createElement(_Grid2.default.Cell, {
      dangerouslySetInnerHTML: {
        __html: bodyHTML
      },
      style: style.body
    }) : _react2.default.createElement(
      _Grid2.default.Cell,
      { style: style.body },
      _react2.default.createElement(
        'p',
        null,
        bodyPlainText
      )
    )
  );
}

LangTextLayout.contextTypes = {
  title: _propTypes2.default.string,
  bodyPlainText: _propTypes2.default.string,
  bodyHTML: _propTypes2.default.string
};

exports.default = LangTextLayout;
