// react-emails/lib/layouts/LangDigestLayout.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Grid = require('../elements/Grid');

var _Grid2 = _interopRequireDefault(_Grid);

var _Header = require('../lang/Header');

var _Header2 = _interopRequireDefault(_Header);

var _MentionsGroup = require('../lang/MentionsGroup');

var _MentionsGroup2 = _interopRequireDefault(_MentionsGroup);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function LangDigestLayout(props, _ref) {
  var intents = _ref.intents,
      brand = _ref.brand;

  return _react2.default.createElement(
    _Grid2.default,
    null,
    _react2.default.createElement(_Header2.default, {
      name: 'Lang.ai Digest',
      title: 'Hi,',
      description: 'We\'ve found new mentions about ' + brand.name + ' in Social Media that need your attention:'
    }),
    Object.keys(intents).map(function (key) {
      return _react2.default.createElement(_MentionsGroup2.default, { key: key, name: key, mentions: intents[key] });
    })
  );
}

LangDigestLayout.contextTypes = {
  intents: _propTypes2.default.object,
  brand: _propTypes2.default.object
};

exports.default = LangDigestLayout;
