// react-emails/lib/layouts/ResetPasswordLayout.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Title = require('../elements/Title');

var _Title2 = _interopRequireDefault(_Title);

var _Grid = require('../elements/Grid');

var _Grid2 = _interopRequireDefault(_Grid);

var _Link = require('../elements/Link');

var _Link2 = _interopRequireDefault(_Link);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var RESET_PASSWORD_URL = 'https://sentisis.io/reset-password';

var styles = {
  title: {
    marginBottom: '10px',
    color: '#10D4AB',
    fontSize: '26px'
  },

  paragraph: {
    fontSize: '16px',
    color: '#7F9BB5',
    lineHeight: '140%'
  },

  greeting: {
    fontSize: '16px',
    color: '#7F9BB5',
    paddingTop: '25px',
    lineHeight: '140%'
  }
};

var ResetPasswordLayout = function (_React$Component) {
  _inherits(ResetPasswordLayout, _React$Component);

  function ResetPasswordLayout() {
    _classCallCheck(this, ResetPasswordLayout);

    return _possibleConstructorReturn(this, (ResetPasswordLayout.__proto__ || Object.getPrototypeOf(ResetPasswordLayout)).apply(this, arguments));
  }

  _createClass(ResetPasswordLayout, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        _Grid2.default,
        null,
        _react2.default.createElement(
          _Title2.default,
          { className: 'Title--xs-small', style: styles.title },
          '\xA1A todos se nos ha olvidado alguna vez! ',
          _react2.default.createElement(
            'span',
            { 'aria-label': ':)', role: 'img' },
            '\uD83D\uDE0A'
          )
        ),
        _react2.default.createElement(
          _Grid2.default.Cell,
          null,
          _react2.default.createElement(
            'p',
            { style: styles.paragraph },
            'Para restablecer tu contrase\xF1a haz click aqu\xED:'
          )
        ),
        _react2.default.createElement(
          _Grid2.default.Cell,
          null,
          _react2.default.createElement(
            _Link2.default,
            { button: true, to: this.buttonHref, className: 'ResetPasswordLayout__button' },
            'Elegir nueva contrase\xF1a'
          )
        ),
        _react2.default.createElement(
          _Grid2.default.Cell,
          { style: styles.greeting },
          _react2.default.createElement(
            'span',
            null,
            'Saludos,'
          ),
          _react2.default.createElement('br', null),
          _react2.default.createElement(
            'span',
            null,
            'Equipo S\xE9ntisis'
          )
        )
      );
    }
  }, {
    key: 'buttonHref',
    get: function get() {
      var token = this.context.token;

      return RESET_PASSWORD_URL + '?token=' + encodeURIComponent(token);
    }
  }]);

  return ResetPasswordLayout;
}(_react2.default.Component);

ResetPasswordLayout.contextTypes = {
  token: _propTypes2.default.string
};
exports.default = ResetPasswordLayout;
