// react-emails/lib/layouts/UserCreatedLayout.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Title = require('../elements/Title');

var _Title2 = _interopRequireDefault(_Title);

var _Grid = require('../elements/Grid');

var _Grid2 = _interopRequireDefault(_Grid);

var _Link = require('../elements/Link');

var _Link2 = _interopRequireDefault(_Link);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var CREATE_USER_URL = 'https://sentisis.io/create-user';

var styles = {
  title: {
    marginBottom: '10px',
    color: '#10D4AB',
    fontSize: '26px'
  },

  paragraph: {
    fontSize: '16px',
    color: '#7F9BB5',
    lineHeight: '140%'
  },

  subtitle: {
    fontSize: '15px',
    color: '#7F9BB5',
    lineHeight: '140%',
    fontWeight: 'bold',
    textAlign: 'left',
    paddingBottom: '20px'
  },

  disclamer: {
    padding: '20px 0',
    fontSize: '13px',
    textAlign: 'left',
    color: '#A2B6C9',
    lineHeight: '140%'
  },

  greeting: {
    fontSize: '16px',
    color: '#7F9BB5',
    paddingTop: '25px',
    lineHeight: '140%'
  }
};

var UserCreatedLayout = function (_React$Component) {
  _inherits(UserCreatedLayout, _React$Component);

  function UserCreatedLayout() {
    _classCallCheck(this, UserCreatedLayout);

    return _possibleConstructorReturn(this, (UserCreatedLayout.__proto__ || Object.getPrototypeOf(UserCreatedLayout)).apply(this, arguments));
  }

  _createClass(UserCreatedLayout, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        _Grid2.default,
        null,
        _react2.default.createElement(
          _Title2.default,
          { className: 'Title--xs-small', style: styles.title },
          '\xA1Bienvenid@ a S\xE9ntisis!'
        ),
        _react2.default.createElement(
          _Grid2.default.Cell,
          null,
          _react2.default.createElement(
            'p',
            { style: styles.paragraph },
            'Enhorabuena, tu empresa ahora cuenta con S\xE9ntisis para extraer el m\xE1ximo valor al an\xE1lisis de redes sociales. A partir de este momento, tomar decisiones basadas en la conversaci\xF3n digital te ser\xE1 m\xE1s f\xE1cil, r\xE1pido e intuitivo.'
          )
        ),
        _react2.default.createElement(
          _Grid2.default,
          { className: 'UserCreatedLayout__wrapper' },
          _react2.default.createElement(
            'tr',
            null,
            _react2.default.createElement(
              'p',
              { className: 'UserCreatedLayout__subtitle', style: styles.subtitle },
              'S\xF3lo 15 segundos m\xE1s para completar tu perfil y acceder al dashboard de an\xE1lisis. Crea tu contrase\xF1a en el siguiente link:'
            ),
            _react2.default.createElement(
              _Link2.default,
              { className: 'UserCreatedLayout__button', button: true, to: this.buttonHref },
              'Completar mi perfil'
            )
          ),
          _react2.default.createElement(
            'tr',
            null,
            _react2.default.createElement(
              'p',
              { className: 'UserCreatedLayout__disclamer', style: styles.disclamer },
              'Este enlace es v\xE1lido por 5 d\xEDas. Si la invitaci\xF3n ha expirado, ponte en contacto con nuestro equipo de soporte ',
              _react2.default.createElement(
                _Link2.default,
                { simple: true, blue: true, to: "mailto: support@sentisis.com" },
                'support@sentisis.com'
              ),
              '. \xA1Gracias!'
            )
          )
        ),
        _react2.default.createElement(
          _Grid2.default.Cell,
          { style: styles.greeting },
          _react2.default.createElement(
            'span',
            null,
            '\xA1Empezamos!, ',
            _react2.default.createElement(
              'span',
              { 'aria-label': 'rocket', role: 'img' },
              '\uD83D\uDE80'
            )
          ),
          _react2.default.createElement('br', null),
          _react2.default.createElement(
            'span',
            null,
            'Equipo S\xE9ntisis'
          )
        )
      );
    }
  }, {
    key: 'buttonHref',
    get: function get() {
      var token = this.context.token;
      var email = this.context.user.email;

      return CREATE_USER_URL + '?token=' + encodeURIComponent(token) + '&email=' + email;
    }
  }]);

  return UserCreatedLayout;
}(_react2.default.Component);

UserCreatedLayout.contextTypes = {
  token: _propTypes2.default.string,
  user: _propTypes2.default.shape({
    email: _propTypes2.default.string
  })
};
exports.default = UserCreatedLayout;
